directory/datadog_checks/directory/directory.py

```diff
@@ -46,12 +50,21 @@ def check(self, _):
                 "Either directory '{}' doesn't exist or the Agent doesn't "
                 "have permissions to access it, skipping.".format(self.config.abs_directory)
             )
-
-            if not self.config.ignore_missing:
+            # report missing directory and skip check
+            if self.config.report_missing:
+                self.log.info('service check SERVICE_DIRECTORY_EXISTS')
+                self.service_check(name=SERVICE_DIRECTORY_EXISTS, status=self.WARNING, message=msg)
+            # or ignore it
+            elif self.config.ignore_missing:
```

Comment:
I guess technically both ignore_missing and report_missing can be true in which case the expected behaviour should be: raise the service check but not an exception?

Reply:
Yes, and here my point was to give higher priority to `report_missing`. Plus, if `report_missing` is true, I though users would be ok with not having the exception, but only a service check reporting the missing dir. To me it was misleading to have both `ignore` and `report` on. What do you think? Better to have only `ignore` to rule on the exception and `report` on the service check?

Reply:
I thinks options are:

1. Not add a new config option: change behaviour from raising an exception to sending the service check. It is a breaking change but in reality not that much because the only way to _consume_ exceptions atm is looking at logs. I think this is the simplest option

2. Add a new option:
2.1: They're independant so one does not affect the others values
2.2: They're not independant: It must be documented on the conf.yaml
2.3: They have incompatible values so a ConfigError is thrown at init if they are set in an incompatible way.

Reply:
Agreed on 1